### ISSUE TITLE
stuck in loop when dataset count < itemsPerPage?

### ISSUE THREAD

**Issue by codemonkeynorth** [deltaepsilon/firebase-paginator#4]
as far as i can tell this will get stuck in a loop if your dataset is smaller than itemsPerPage

assuming `keys.length = 3`, and `pageSize=10`

```javascript
 if (keys.length === pageSize + 1) {  // false
  if (isLastPage) {
    delete collection[keys[keys.length - 1]];
  } else {
     delete collection[keys[0]];
  }
} else if (isForward) {  // false for default first page, gets called on second loop from last condition
  return setPage(); // this will loop us, fail the length check and then hit the condition below
} else {  // we will get here first time, then hit the isForward condition in the loop, then loop back here?
  return setPage(undefined, true, true);
}
```

thanks
J

**Comment by Caerbannog** [comment]
confirmed

**Comment by AlecAnthonyGarcia** [comment]
@deltaepsilon Also confirmed. If the dataset that is being paginated over is less than the page size, it will fail to return. So if my page size is 10 and I only have 3 items, I'll never get them.

**Comment by deltaepsilon** [comment]
Sorry I haven't addressed any of the issues on this project yet. I had a bad email filter that was suppressing GitHub issue alerts.

I'll take a crack at fixing these issues tomorrow when I get more time at my desk.

**Comment by deltaepsilon** [comment]
Fixed! 01be73d5c8794e93d05f2d6ea1f3cda59cc98778
